monailabel/utils/others/post.py

```diff
@@ -96,7 +96,7 @@ def __call__(self, data):
                 resizer = Resize(spatial_size=spatial_size, mode=self.mode[idx])
                 result = resizer(result, mode=self.mode[idx], align_corners=self.align_corners[idx])
 
-            d[key] = result if len(result.shape) <= 3 else result[0]
+            d[key] = result if len(result.shape) <= 3 else result[0] if result.shape[0] == 1 else result
```

Comment:
@SachidanandAlle I had to add this line in order to allow logits with dimension N, H, W, D to be written back to data dictionary. Maybe there is a better way or perhaps I missed something here? As this is common code with other methods so better to check if we are not breaking something here. Could you have a look please?

Reply:
best way to check/verify is run some deepgrow 2D/3D actions in slicer

Reply:
verified